presto-hive/src/main/java/com/facebook/presto/hive/FilteringPageSource.java

```diff
@@ -119,6 +122,19 @@ public FilteringPageSource(
         }
         else {
             RowExpression expression = replaceExpression(optimizedRemainingPredicate, variableToInput);
+
+            DynamicFilters.DynamicFilterExtractResult extractDynamicFilterResult = extractDynamicFilters(expression);
+
+            // Extract the static ones
+            // TODO: james, and() is a short cut hack
```

Comment:
Remove the comment; just leave the following line. Add a comment: "dynamic filter will be added through subfield pushdown"